File: instructor Henry/lenguajes/Python/orientaado-a-objetos/05-08-2024.py
class Transporte:
    def __init__(self, marca=None):
        self.marca = marca

class Terrestre(Transporte):
    def __init__(self, marca=None, llantas=None):
        super().__init__(marca)
        self.llantas = llantas

class Aereo(Transporte):
    def __init__(self, marca=None, motores=None):
        super().__init__(marca)
        self.motores = motores

class Auto(Terrestre):
    def __init__(self, marca=None, llantas=None, aire=None):
        super().__init__(marca, llantas)
        self.aire = aire

class Moto(Terrestre):
    def __init__(self, marca=None, llantas=None, cascos=None):
        super().__init__(marca, llantas)
        self.cascos = cascos

class Avion(Aereo):
    @staticmethod
    def manual():
        print("hola")

# Ejecución del programa
carro = Auto(marca="Ford", llantas=4, aire=True)

moto = Moto(marca="Suzuki", llantas=2, cascos=2)

avion = Avion(marca="Delta", motores=3)
Avion.manual()

File: instructor Henry/lenguajes/Ruby/orientado-a-objetos/05-08-2024.rb
class Transporte
    attr_accessor :marca
  end
  
  class Terrestre < Transporte
    attr_accessor :llantas
  end
  
  class Aereo < Transporte
    attr_accessor :motores
  end
  
  class Auto < Terrestre
    attr_accessor :aire
  end
  
  class Moto < Terrestre
    attr_accessor :cascos
  end
  
  class Avion < Aereo
    def self.manual
      puts "hola"
    end
  end
  
  # Ejecución del programa
  carro = Auto.new
  carro.marca = "Ford"
  carro.llantas = 4
  carro.aire = true
  
  moto = Moto.new
  moto.marca = "Suzuki"
  moto.llantas = 2
  moto.cascos = 2
  
  avion = Avion.new
  avion.marca = "Delta"
  avion.motores = 3
  Avion.manual
  